File: app/src/main/java/com/bypassmobile/octo/BaseActivity.java
package com.bypassmobile.octo;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;

import com.bypassmobile.octo.core.rest.GithubEndpoint;

import retrofit.RestAdapter;

public class BaseActivity extends AppCompatActivity {

    private AlertDialog mProgressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onStop() {
        dismissProgressDialog();
        super.onStop();
    }

    protected void showProgressDialog(@NonNull String message) {
        if (mProgressDialog != null && mProgressDialog.isShowing()) {
            mProgressDialog.dismiss();
        }

        mProgressDialog = new ProgressDialog.Builder(this)
                .setMessage(message)
                .setCancelable(false)
                .show();
    }

    protected void dismissProgressDialog() {
        if (mProgressDialog != null) {
            mProgressDialog.dismiss();
        }
    }
}

File: app/src/main/java/com/bypassmobile/octo/BaseActivity.kt
package com.bypassmobile.octo

import android.app.Activity
import android.os.Bundle

import com.bypassmobile.octo.rest.GithubEndpoint
import com.google.gson.FieldNamingPolicy
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import io.reactivex.schedulers.Schedulers

import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

open class BaseActivity : Activity() {

    var endpoint: GithubEndpoint? = null
    var gson: Gson? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        gson = GsonBuilder()
                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
                .create()

        endpoint = Retrofit.Builder()
                .baseUrl(GithubEndpoint.SERVER)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.createWithScheduler(Schedulers.io()))
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build().create(GithubEndpoint::class.java)
    }
}
